File: fsharp-introduction/WhyFSharp/src/CSharp/Conciseness.cs
using System;
using System.Linq;

namespace CSharp
{
    public static class Conciseness
    {
        public class Person
        {
            public string Name { get; set; }
            public int Age { get; set; }
        }

        public static void PrintPerson(Person person)
        {
            int age = person.Age;
            string name = person.Name;

            Console.WriteLine(String.Join(",", Enumerable.Range(0, age).Select(a => $"{name} -- {a}")));
        }
    }
}

File: fsharp-introduction/WhyFSharp/src/FSharp/Conciseness.fs
namespace FSharp

open System

module Conciseness =

    type Person = { Name : string; Age : int; }
    type Animal = { Color : string; Age : int; }
    
    let printPerson (person:Person) =
        let (name, age) = ("some name", person.Age)
        [0..age]
        |> List.map (fun i -> sprintf "%s -- %i" name i) 
        |> fun names -> String.Join(",", names)
        |> printfn "%s"